cmd/manager/main.go

```diff
@@ -554,27 +557,51 @@ func startOperator(ctx context.Context) error {
 
 	// configure the manager cache based on the number of managed namespaces
 	managedNamespaces := viper.GetStringSlice(operator.NamespacesFlag)
+
 	switch {
 	case len(managedNamespaces) == 0:
 		log.Info("Operator configured to manage all namespaces")
+		opts.Cache.DefaultNamespaces = map[string]cache.Config{corev1.NamespaceAll: {}}
 	case len(managedNamespaces) == 1 && managedNamespaces[0] == operatorNamespace:
 		log.Info("Operator configured to manage a single namespace", "namespace", managedNamespaces[0], "operator_namespace", operatorNamespace)
+		opts.Cache = cache.Options{
+			ByObject: map[client.Object]cache.ByObject{
+				&storagev1.StorageClass{}: {
```

Comment:
From the docs (which I don't fully understand) 

> This must be unset for cluster-scoped objects.

Is that what you are doing here? Intuitively I would have thought we must not set cluster-scope resources here?

Reply:
Yes, I'm wrong, you're right.